libraries/AP_BoardConfig/px4_drivers.cpp

```diff
@@ -204,52 +202,53 @@ void AP_BoardConfig::px4_setup_sbus(void)
 #endif
 }
 
-
 /*
-  setup CANBUS drivers
+ setup CANBUS drivers
  */
 void AP_BoardConfig::px4_setup_canbus(void)
 {
 #if HAL_WITH_UAVCAN
-    if (px4.can_enable >= 1) {
-        // give time for other drivers to fully start before we start
-        // canbus. This prevents a race where a canbus mag comes up
-        // before the hmc5883
-        hal.scheduler->delay(500);
-        if (px4_start_driver(uavcan_main, "uavcan", "start")) {
-            hal.console->printf("UAVCAN: started\n");            
-            // give some time for CAN bus initialisation
-            hal.scheduler->delay(2000);
-        } else {
-            hal.console->printf("UAVCAN: failed to start\n");
+    if (_var_info_can._can_enable >= 1) {
```

Comment:
It was just a matter of setting *_st_can_enable* before calling *px4_setup*, but OK, this is just in initialization, it's not that bad.